browser/ui/views/side_panel/brave_bookmarks_side_panel_view.cc

```diff
@@ -106,23 +104,39 @@ class BookmarksSidePanelHeaderView : public views::View {
 
 }  // namespace
 
-BraveBookmarksSidePanelView::BraveBookmarksSidePanelView(Browser* browser) {
+BraveBookmarksSidePanelView::BraveBookmarksSidePanelView(Browser* browser)
+    : browser_(*browser) {
   SetLayoutManager(std::make_unique<views::FlexLayout>())
       ->SetOrientation(views::LayoutOrientation::kVertical);
-  AddChildView(std::make_unique<BookmarksSidePanelHeaderView>(browser));
-  AddChildView(std::make_unique<views::Separator>())
-      ->SetColorId(kColorSidebarPanelHeaderSeparator);
 
   // Reuse upstream's bookmarks panl nwebui.
-  auto* web_view =
-      AddChildView(BookmarksSidePanelCoordinator::GetOrCreateForBrowser(browser)
-                       ->CreateBookmarksWebView());
+  auto* web_view = AddChildView(
+      BookmarksSidePanelCoordinator::GetOrCreateForBrowser(&(browser_.get()))
+          ->CreateBookmarksWebView());
   web_view->SetProperty(
       views::kFlexBehaviorKey,
       views::FlexSpecification(views::MinimumFlexSizeRule::kPreferred,
                                views::MaximumFlexSizeRule::kUnbounded));
-
+  UpdateNativeUI();
   StartObservingWebWebViewVisibilityChange(web_view);
 }
 
 BraveBookmarksSidePanelView::~BraveBookmarksSidePanelView() = default;
+
+void BraveBookmarksSidePanelView::UpdateNativeUI() {
+  if (header_view_) {
+    RemoveChildView(header_view_);
```

Comment:
Or you might want to just reattach the old view instead of creating a new one. (Assuming that that's how we simulate PropgateThemeChanged() when native theme changes.)